qiskit_nature/circuit/library/ansatzes/evolved_operator_ansatz.py

```diff
@@ -0,0 +1,149 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The evolved operator ansatz."""
+
+from typing import List, Union, Optional
+
+from qiskit.circuit import Parameter, ParameterVector, QuantumRegister
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit.opflow import OperatorBase, EvolutionBase, PauliTrotterEvolution
+
+
+class EvolvedOperatorAnsatz(BlueprintCircuit):
+    """The evolved operator ansatz."""
+
+    def __init__(self,
+                 operators: Optional[Union[OperatorBase, List[OperatorBase]]] = None,
+                 reps: int = 1,
+                 evolution: Optional[EvolutionBase] = None,
+                 insert_barriers: bool = False,
```

Comment:
Shouldn't this have a initial state (QuantumCircuit) here too in the base class as this should be directly usable if you have your own set of operators for this.